include/GafferScene/BranchCreator.h

```diff
@@ -56,6 +56,26 @@ IE_CORE_FORWARDDECLARE( StringPlug )
 namespace GafferScene
 {
 
+/// A base class to simplify the process of creating new branches in the scene
+/// hierarchy. The source data for each branch is specified by the input
+/// locations matched by the filter. By default, the branches are made
+/// underneath the source locations, but they can be relocated by using the
+/// `destination` plug. We use the following terminology :
+///
+/// - `sourcePath` : An input location matched by `filter`. Each source will create
+///   exactly one branch.
+/// - `destinationPath` : The location where branch will be created in the output scene.
+///   This is specified by `destinationPlug()`, and defaults to `sourcePath`. Multiple
+///   branches may have the same destination, but this is handled transparently and
```

Comment:
The transparent handling is that the branches will get an incremental number appended, correct?  Probably worth specifying what actually happens here.

Reply:
Correct, clarified in https://github.com/GafferHQ/gaffer/pull/4200/commits/22df375f13dd3ddf10e2e429da957a93d7bcb555.